[RFC] fix: useSprings controller clear refs properly when length changes in React.StrictMode #1532

tarngerine opened the pull request:
### Why
fixes https://github.com/pmndrs/react-spring/issues/1532#issuecomment-870784013

### What
RFC

### Checklist

- [ ] Documentation updated (N/A)
- [ ] Demo added
- [ ] Ready to be merged

joshuaellis commented on demo/src/index.tsx:
```diff
@@ -3,4 +3,11 @@ import ReactDOM from 'react-dom'
+  document.getElementById('root')
+)
+
+// ReactDOM.render(<App />, document.getElementById('root'))
```
Lets get rid of this comment. It's probably better to make sure the library works in Strict Mode.

joshuaellis commented on demo/src/App.jsx:
```diff
@@ -52,6 +53,7 @@ const links = {
   'parallax-vert': ParallaxVert,
   'parallax-sticky': ParallaxSticky,
   three: SimpleThree,
+  test: Test,
```
Can we name this something more intuitive, I guess this is all about adding and subtracting springs? So something along that line.

tarngerine replied:
Do you think we should include this demo or just rely on the new tests? It's not really at the calibre of the rest of the demos, and I just grabbed it from a discord user who had posted this bug originally.

tarngerine replied:
From ericvu in #react-spring

tarngerine replied:
i think this demo app is super confusing since it's half-broken, do you think its okay to remove this demo app and just rely on the tests?

joshuaellis reviewed (changes requested):
Looking at the changes, the tests pass, I think if we just do those little changes, we can release this as a patch. Would you be happy to offer some support if this leads to unexpected side effects in people's code? although I'm hoping it wont 🙏🏼

tarngerine commented:
ok, i removed the demo app bc i dont think its necessary as we have implemented the tests, and removed the commented non-strict-mode line

happy to provide any support post launch, altho i will be "out of office" the first week of august

lmk good next steps, thanks!

joshuaellis commented:
Hey @tarngerine sorry, I've been so snowed under with work at the moment. Yeah I think upon reflection, you're probably right, let's remove the app and keep the tests, thats the point of tests after all?
